docs/guide/snaps-rpc-api.md

```diff
@@ -596,3 +596,70 @@ await wallet.request({
   params: ['clear'],
 });
 ```
+
+### `snap_notify`
+
+::: warning Only Callable By
+
+- Snaps
+  :::
+
+#### Parameters
+
+```typescript
+interface SnapNotifyParams {
+  /**
+   * Enum type to determine notification type.
+   */
+  type: NotificationType;
+
+  /**
+   * A message to show on the notification.
+   */
+  message: string;
+}
+```
+
+- `Array`
+
+  0. `SnapNotifyParams` - An object containing the contents of the notification.
+
+#### Notification Type
+
+```typescript
+enum NotificationType {
+  /**
+   * A default browser notification
+   */
+  native = 'native',
+
+  /**
+   * A in-app notification stored in the metamask state.
```

Comment:
I think "An in-app notification showed inside the MetaMask UI" might be better, remove the line below.